cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -353,8 +353,6 @@ func (s *Server) Run(queue *Queue) {
 		metricIndexDuration.WithLabelValues(string(state)).Observe(time.Since(start).Seconds())
 		if err != nil {
 			log.Printf("error indexing %s: %s", args.String(), err)
-			queue.SetLastIndexFailed(name)
-			continue
```

Comment:
With this change `err` becomes obsolete, doesn't it? All information of what happened is carried in `state`, which is always assumed to be well defined?

Reply:
good point! will update.

Reply:
Returning the error is super convenient since we capture it in a defer. I think the logs are actually useful though, so gonna leave it in for now since it will require a bit more refactoring than I am comfortable with.